File: gvpr/Java/GvprParserBase.java
import org.antlr.v4.runtime.*;

public abstract class GvprParserBase extends Parser
{
    protected GvprParserBase(TokenStream input)
    {
        super(input);
    }

    protected boolean IsSemiRequired()
    {
	var c = this._input.LT(-1);
        var d = this._input.LT(1);
        return c.getType() != gvprParser.CCBC;
    }

    protected boolean IsSemiNotRequired()
    {
	var c = this._input.LT(-1);
	var d = this._input.LT(1);
        return c.getType() == gvprParser.CCBC;
    }
}

File: gvpr/Cpp/GvprParserBase.cpp
#include "gvprParser.h"

using namespace antlr4;

bool GvprParserBase::IsSemiRequired()
{
	auto c = this->getTokenStream()->LT(-1);
	auto d = this->getTokenStream()->LT(1);
	return c->getType() != gvprParser::CCBC;
}

bool GvprParserBase::IsSemiNotRequired()
{
	auto c = this->getTokenStream()->LT(-1);
	auto d = this->getTokenStream()->LT(1);
	return c->getType() == gvprParser::CCBC;
}
